### PR TITLE
Fix LocationIndicator

### PR DESCRIPTION
**Pull request by galinelle**
This changes image size properties to be scales, tests for expressions
in paint properties, moved image-tilt-displacement and perspective-compensation
into paint properties, and some minor cleanup.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [x] document any changes to public APIs

### CODE REVIEW

File: src/mbgl/renderer/layers/render_location_indicator_layer.cpp [hunk 1]
```diff
@@ -363,7 +370,7 @@ void main() {
 
     void release() {
         if (!simpleShader.program) return;
-        textures.clear();
+        for (auto t : textures) t.second->release();
```
Comment (tmpsantos) [inline src/mbgl/renderer/layers/render_location_indicator_layer.cpp:373]:
```
loop variable is copied but only used as const reference; consider making it a const reference [performance-for-range-copy,-warnings-as-errors]
        for (auto t : textures) t.second->release();
                  ^
             const  &
```

File: src/mbgl/renderer/layers/render_location_indicator_layer.cpp [hunk 2]
```diff
@@ -539,10 +546,12 @@ void main() {
                 params.perspectiveCompensation; // Compensation factor for the perspective deformation
         //     ^ clamping this to 0.8 to avoid growing the puck too much close to the camera.
         const double shadowRadius =
-            params.puckShadowSizePx * M_SQRT2 * 0.5 *
+            ((texShadow) ? texShadow->width : 0.0) * params.puckShadowScale * M_SQRT2 * 0.5 *
```
Comment (alexshalamov) [inline src/mbgl/renderer/layers/render_location_indicator_layer.cpp:549]: Could you test with '@2x' and '@3x' images? We need to scale quads based on image's pixel ratio.

Comment (galinelle) [reply src/mbgl/renderer/layers/render_location_indicator_layer.cpp]: added support for dividing the width by the image pixel ratio

File: metrics/tests/location_indicator/image_pixel_ratio/style.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          }
+        ]
+      }
+    }
+  },
+  "center": [ 139.766707, 35.693055 ],
+  "zoom": 16,
+  "pitch" : 0,
+  "bearing" : 42,
+  "sources": {},
```
Comment (alexshalamov) [inline metrics/tests/location_indicator/image_pixel_ratio/style.json:57]: sources defined twice

**Pull request merged**